pdns/dnsname.hh

```diff
@@ -371,49 +371,75 @@ struct SuffixMatchTree
    anything part of that domain will return 'true' in check */
 struct SuffixMatchNode
 {
-  SuffixMatchNode()
-  {}
-  SuffixMatchTree<bool> d_tree;
+  public:
+    SuffixMatchNode()
+    {}
+    SuffixMatchTree<bool> d_tree;
+
+    void add(const DNSName& dnsname)
+    {
+      d_tree.add(dnsname, true);
+      d_nodes.insert(dnsname);
+      updateHuman();
```

Comment:
I'm a bit worried that we are now doing an `O(n)` operation on insertion, although that seems fine with the way we currently use `SuffixMatchNode`, except perhaps for the `SuffixMatchNodeRule` in dnsdist where it depends how it's used from `Lua`.